src/helpers/grapher.js

```diff
@@ -0,0 +1,258 @@
+  var buttons = [];
+  var metricLines = []; // list of react components
+  var metricStatsNames = Object.keys(
+    props.data[currentTestCase][props.metric][0]
+  ); // names of stats
+  var mainColour = props.idx % 2; // main colour for that metric., red , green ,blue
+  var i = 0;
+  var basicVisibility = {};
+  metricStatsNames.forEach((name) => {
+    basicVisibility[name] = true;
+  }); //init  stats visibility
+  const [visibility, setVisibility] = useState(basicVisibility);
+
+  metricStatsNames.forEach((name) => {
+    if (!IGNORE_ATTRIBUTES_GRAPH.includes(name)) {
+      metricLines.push(
+        <Line
+          type="monotone"
+          dataKey={name}
+          stroke={getRandomColour(mainColour, i)}
+          legendType={visibility[name] ? "circle" : "diamond"}
+          style={{ display: visibility[name] ? "block" : "none" }}
+          activeDot={{ r: visibility[name] ? 7 : 0 }}
+        >
+          {name.includes("Average") ? (
+            <ErrorBar
+              dataKey="Std"
+              direction="y"
+              style={{ display: visibility[name] ? "block" : "none" }}
+            />
+          ) : (
+            <></>
+          )}
+        </Line>
+      );
+      //Draw each stats
+
+      i++;
+    }
+  });
+  if (props.config["metricConfig"][props.metric].thresholds !== undefined) {
+    metricLines.push(
+      <ReferenceLine
+        y={props.config["metricConfig"][props.metric].thresholds}
+        label="Threshold"
+        stroke="blue"
+        strokeDasharray="10 10"
+      />
+    ); //Add a threshold line
+  }
+  testVariables.forEach((varSet, i) => {
+    var options = [];
+    varSet.forEach((value) => {
+      options.push(<option>{value}</option>);
+    });
+
+    buttons.push(
+      <div class="select_box">
+        <label>{TEST_VARIABLES[i]}</label>
+        <select
+          id={`testCase-${props.title}-${i}`}
+          onChange={() => {
+            var testCase = "";
+            var n_variables = testVariables.length;
+            for (var j = 0; j < n_variables; j++) {
+              testCase += document.getElementById(
+                `testCase-${props.title}-${j}`
+              ).value;
+              if (j < n_variables - 1) {
+                testCase += "-";
+              }
+            }
+            setCurrentTest(testCase);
+          }}
+        >
+          {options}
+        </select>
+      </div>
+    );
+  });
+
+  props.data[currentTestCase][props.metric].forEach((item) => {
+    if (item["isRelease"]) {
+      metricLines.push(
+        <ReferenceLine
+          x={item["Hash"]}
+          label=""
+          stroke="#8B008B"
+          strokeWidth={3}
+        />
+      );
+    }
+  });
+  var size = parseInt(props.config.graphSize);
+
+  return (
+    <div class="graph">
+      <div class="button_container">{buttons}</div>
+      <h2>{`${props.title}-${currentTestCase}`}</h2>
+      <LineChart
+        width={487.5 + 162.5 * size}
+        height={300 + 75 * size}
```

Comment:
What is the reason for these specific dimensions?

Reply:
I will add a comment on it. But basically i tested the website on the amazon monitors, and my work laptop. Then I did a linear regression between these two; so that max graph size would look great on the monitor, and graph size 3 looks good on the laptop. But I should have been more clear with a comment